### PR TITLE
Fetch products based on category

### PR DESCRIPTION
**Pull request by msaivasanth**
Added a feature that helps in showing the details of products of same category only.

### CODE REVIEW

File: src/Components/Search.tsx [hunk 1]
```diff
@@ -47,9 +47,10 @@ const Search = () => {
         }
       }
       
-      const debounceFn = myDebounce((value1: string) => {
-        const suggestions = searchSuggestions(value1)
-        setSugg(suggestions?.slice(0, 10))
+      const debounceFn = myDebounce(async (value1: string) => {
```
Comment (syedzeeshan1) [inline src/Components/Search.tsx:50]: can you tell me why you had to use async await here?

File: src/ProductsContext/ProductState.tsx [hunk 2]
```diff
@@ -253,6 +258,7 @@ const ProductState: React.FC<ProductStateProps> = (props: any) => {
         if(product.title.toLowerCase().indexOf(search.toLowerCase()) !== -1 || product.description.toLowerCase().indexOf(search.toLowerCase()) !== -1) return product;
       })
       setProducts(result)
+      console.log(products)
```
Comment (syedzeeshan1) [inline src/ProductsContext/ProductState.tsx]: nit: remove

**Review by syedzeeshan1 — COMMENTED** [review]
Overall LGTM!

**Comment by syedzeeshan1** [comment]
> Overall LGTM!

can you give me access everytime with MR?

File: src/Components/Search.tsx (hunk 1 above)
Comment (msaivasanth) [reply src/Components/Search.tsx:50]: Here I called a async function. That's why I used async await.

File: src/ProductsContext/ProductState.tsx (hunk 2 above)
Comment (msaivasanth) [reply src/ProductsContext/ProductState.tsx]: Ok sir, I will remove it.

**Comment by msaivasanth** [comment]
> > Overall LGTM!
> 
> can you give me access everytime with MR?


Yes sir,
For now this is for current MR,

https://product-inventory-ysce-git-feature-5d93ba-msaivasanths-projects.vercel.app?_vercel_share=0xmkbmoYaleugp1TB5LulPFv71RQFMiS

with this link you can access the MR in vercel

**Review by syedzeeshan1 — APPROVED** [review]

**Pull request merged**